### ISSUE TITLE
Inconsistent basis set parsing

### ISSUE THREAD

**Issue by ferchault** [psi4/psi4#1996]
When specifying a custom basis set (6-31G in this case), the parser accepts "D" as a label for scientific notation for the first two of three columns only. If "D" is present in the third column, the parser fails. This is because `psi4/driver/qcdb/libmintsbasissetparser.py` limits the number of replacements of "D" with "e" to two per line. Is there a particular reason for this limit? Otherwise I'm happy to submit a PR fixing this.

Fails:
```
molecule mol {
N 0.0 0.0 0.0
N 0.0 0.0 1.0
0 1
}

basis {
    assign genbas
[genbas]
cartesian

****
N     0
S   6   1.00
      0.4173511460D+04       0.1834772160D-02
      0.6274579110D+03       0.1399462700D-01
      0.1429020930D+03       0.6858655181D-01
      0.4023432930D+02       0.2322408730D+00
      0.1282021290D+02       0.4690699481D+00
      0.4390437010D+01       0.3604551991D+00
SP   3   1.00
      0.1162636186D+02      -0.1149611817D+00       0.6757974388D-01
      0.2716279807D+01      -0.1691174786D+00       0.3239072959D+00
      0.7722183966D+00       0.1145851947D+01       0.7408951398D+00
SP   1   1.00
      0.2120314975D+00       0.1000000000D+01       0.1000000000D+01
****

}
e, wfn = energy('scf', return_wfn=True)
```
with message:
```
Traceback (most recent call last):
  File "/mnt/c/Users/guido/workcopies/psi4/objdir/stage/bin/psi4", line 337, in <module>
    exec(content)
  File "<string>", line 49, in <module>
  File "/mnt/c/Users/guido/workcopies/psi4/objdir/stage/lib/psi4/driver/driver.py", line 576, in energy
    wfn = procedures['energy'][lowername](lowername, molecule=molecule, **kwargs)
  File "/mnt/c/Users/guido/workcopies/psi4/objdir/stage/lib/psi4/driver/procrouting/proc.py", line 2250, in run_scf
    scf_wfn = scf_helper(name, post_scf=False, **kwargs)
  File "/mnt/c/Users/guido/workcopies/psi4/objdir/stage/lib/psi4/driver/procrouting/proc.py", line 1435, in scf_helper
    base_wfn = core.Wavefunction.build(scf_molecule, core.get_global_option('BASIS'))
  File "/mnt/c/Users/guido/workcopies/psi4/objdir/stage/lib/psi4/driver/p4util/python_helpers.py", line 122, in _core_wavefunction_build
    basis = core.BasisSet.build(mol, "ORBITAL", basis)
  File "/mnt/c/Users/guido/workcopies/psi4/objdir/stage/lib/psi4/driver/p4util/python_helpers.py", line 87, in _pybuild_basis
    return_atomlist=return_atomlist)
  File "/mnt/c/Users/guido/workcopies/psi4/objdir/stage/lib/psi4/driver/qcdb/libmintsbasisset.py", line 680, in pyconstruct
    verbose=verbose)
  File "/mnt/c/Users/guido/workcopies/psi4/objdir/stage/lib/psi4/driver/qcdb/libmintsbasisset.py", line 862, in construct
    shells, msg, ecp_shells, ecp_msg, ecp_ncore = parser.parse(entry, lines)
  File "/mnt/c/Users/guido/workcopies/psi4/objdir/stage/lib/psi4/driver/qcdb/libmintsbasissetparser.py", line 304, in parse
    contraction = float(what.group(3))

ValueError: could not convert string to float: '0.6757974388D-01'

Printing out the relevant lines from the Psithon --> Python processed input file:
    ****
    """
        return basstrings
    qcdb.libmintsbasisset.basishorde['ANONYMOUSA2D5A0E2'] = basisspec_psi4_yo__anonymousa2d5a0e2
    core.set_global_option("BASIS", "anonymousa2d5a0e2")
--> e, wfn = energy('scf', return_wfn=True)


!--------------------------------------------------------!
!                                                        !
!  could not convert string to float: '0.6757974388D-01' !
!                                                        !
!--------------------------------------------------------!
```

**Comment by loriab** [comment]
you're right, that should definitely be a `replace(..., 'e', 3)` for the sp case. PR would be great!

**Comment by ferchault** [comment]
fixed in 6a5434695c6abff861738252b7acf47e1aca0a65
